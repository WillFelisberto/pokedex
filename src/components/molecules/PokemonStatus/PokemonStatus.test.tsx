import { render, screen } from '@/tests/test-utils';

import { PokemonStatus } from '.';

describe('<PokemonStatus />', () => {
  it('should render the correct values', () => {
    const { container } = render(<PokemonStatus height={10} weight={100} />);

    // Verifica se os elementos contendo os textos convertidos est√£o presentes
    expect(screen.getByText(/10 kg/i)).toBeInTheDocument();
    expect(screen.getByText(/1 m/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
