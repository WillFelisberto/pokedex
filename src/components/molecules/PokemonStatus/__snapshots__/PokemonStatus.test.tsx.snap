// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonStatus /> should render the correct values 1`] = `
<div>
  <div
    class="grid grid-cols-2 gap-2"
  >
    <div
      class="flex flex-col gap-[4.5px] items-left  p-3 rounded-lg  "
    >
      <div
        class="flex items-end gap-[8.5px] text-gray-500 text-xs font-semibold uppercase"
      >
        <img
          alt="weight"
          height="18"
          src="/icons/svg/outlined/weight.svg"
          width="18"
        />
        <span>
          Peso
        </span>
      </div>
      <div
        class="mt-1 font-(--font-poppins)  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]"
      >
        10 kg
      </div>
    </div>
    <div
      class="flex flex-col gap-[4.5px] items-left  p-3 rounded-lg  "
    >
      <div
        class="flex items-end gap-[8.5px] text-gray-500 text-xs font-semibold uppercase"
      >
        <img
          alt="height"
          height="18"
          src="/icons/svg/outlined/height.svg"
          width="18"
        />
        <span>
          Altura
        </span>
      </div>
      <div
        class="mt-1 font-(--font-poppins)  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]"
      >
        1 m
      </div>
    </div>
  </div>
</div>
`;
